### PR TITLE
feat: don't depend on releases file for direct mode publishing

### PR DESCRIPTION
**Pull request by ryan-timothy-albert**
Make it so publishing in direct mode does not depend on the publishing file

Temporarily leaving around the old flow for terraform because it currently depends on previous release version from the file

### CODE REVIEW

File: internal/environment/environment.go [hunk 1]
```diff
@@ -193,10 +193,6 @@ func IsDocsGeneration() bool {
 	return strings.Contains(languages, "docs")
 }
 
-func CreateGitRelease() bool {
```
Comment (ryan-timothy-albert) [inline internal/environment/environment.go:196]: This check is pointless. It always evaluates to true. See IsLanguagePublished("swift")

File: internal/actions/runWorkflow.go [hunk 2]
```diff
@@ -259,9 +254,13 @@ func finalize(inputs finalizeInputs) error {
 	case environment.ModeDirect:
 		var releaseInfo *releases.ReleasesInfo
 		if !inputs.SourcesOnly {
-			releaseInfo, err = getReleasesInfo()
-			if err != nil {
-				return err
+			releaseInfo = inputs.currentRelease
+			// TODO: Temporary until we figure out how to handle parsed terraform previous release version
```
Comment (ryan-timothy-albert) [inline internal/actions/runWorkflow.go:258]: @ThomasRooney 
If you have any ideas here

File: internal/actions/release.go [hunk 3]
```diff
@@ -28,7 +28,7 @@ func Release() error {
 
 	dir := "."
 	usingReleasesMd := false
-	// TODO: Maybe instead check for workflow dispatch event
+	// We do not currently support specific target workflow dispatch for terraform. This is because terraform relies on release history.
```
Comment (chase-crumbaugh) [inline internal/actions/release.go:31]: This comment doesn't seem to align with what's going on below? Where is terraform mentioned?

Comment (ryan-timothy-albert) [reply internal/actions/release.go:31]: The workflow_dispatch won't be setup in speakeasy configure publishing for terraform. I can also add an error state here though.

Comment (ryan-timothy-albert) [reply internal/actions/release.go]: Updated. I think it makes more sense now

File: internal/actions/runWorkflow.go (hunk 2 above)
Comment (ThomasRooney) [reply internal/actions/runWorkflow.go:258]: Just remove the TF publishing target -- AFAIK it's entirely unused. Everyone just uses goreleaser.

File: internal/environment/environment.go (hunk 1 above)
Comment (ThomasRooney) [reply internal/environment/environment.go:196]: 😬 thanks for cleaning this up

**Review by chase-crumbaugh — APPROVED** [review]

**Pull request merged**